File: dotnet/niuz.application/services/PublishingService.cs
using niuz.application.entities;
using niuz.application.events;
using niuz.application.repositories;

namespace niuz.application.services
{
    public class PublishingService
    {
        private readonly IAuthorRepository authors;
        private readonly IArticleRepository articles;
        private readonly IPaymentRepository payments;
        private readonly IEventPublisher publisher;

        public PublishingService(IAuthorRepository authors, IArticleRepository articles, IPaymentRepository payments, IEventPublisher publisher)
        {
            this.authors = authors;
            this.articles = articles;
            this.payments = payments;
            this.publisher = publisher;
        }

        public void Publish(string articleId)
        {
            var article = articles.GetByArticleId(articleId);
            var author = authors.GetByAuthorId(article.AuthorId);

            publisher.Publish(new ArticlePublished(article.Headline, author.Name));
            if (author.PaysByPublication)
            {
                payments.Save(new Payment(author.Rate, author.BankAccount, author.Name, article.Headline));    
            }
        }
    }
}

File: java/src/main/java/com/dddeurope/niuz/services/PublishingService.java
package com.dddeurope.niuz.services;

import com.dddeurope.niuz.entities.Article;
import com.dddeurope.niuz.entities.Author;
import com.dddeurope.niuz.events.ArticlePublished;
import com.dddeurope.niuz.events.EventPublisher;
import com.dddeurope.niuz.repositories.ArticleRepository;
import com.dddeurope.niuz.repositories.AuthorRepository;

public class PublishingService {
    private final AuthorRepository authors;
    private final ArticleRepository articles;
    private final EventPublisher publisher;

    public PublishingService(AuthorRepository authors, ArticleRepository articles, EventPublisher publisher) {
        this.authors = authors;
        this.articles = articles;
        this.publisher = publisher;
    }

    public void publish(String articleId) {
        Article article = articles.getByArticleId(articleId);
        Author author = authors.getByAuthorId(article.getAuthorId());

        publisher.publish(new ArticlePublished(author.getId(), author.getName(), article.getHeadline()));
    }
}
